### PR TITLE
Load Channels with `.` in their name

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Allows loading of channels with a period in their name (other than for the file extension)

**How did you implement your changes**

Used `os.path.splitext` instead of `.split` for the path.
Updated dependencies for Python 3.12 support.

**Remaining issues**

N/A.

### CODE REVIEW

**Comment by srivarra** [comment]
@camisowers Yeah I'll add a test. `load_imgs_from_dir` does not have this issue. It uses `io_utils.remove_file_extensions` to remove file extensions, instead of directly splitting the string itself.

File: src/alpineer/load_utils.py [hunk 1]
```diff
@@ -139,7 +139,9 @@ def load_imgs_from_tree(
     # otherwise, fill channel names with correct file extension
     elif not all([img.endswith(tuple(EXTENSION_TYPES["IMAGE"])) for img in channels]):
         # need this to reorder channels back because list_files may mess up the ordering
-        channels_no_delim = [os.path.splitext(img)[0] for img in channels]
+        # channels_no_delim = [os.path.splitext(img)[0] for img in channels]
```
Comment (camisowers) [inline src/alpineer/load_utils.py:142]: This line should be deleted.

**Review by camisowers — CHANGES_REQUESTED** [review]
Looks good! Just one minor fix.

**Review by camisowers — APPROVED** [review]

**Review by alex-l-kong — APPROVED** [review]
Good stuff

**Pull request merged**